transport.go

```diff
@@ -126,7 +119,7 @@ func (wb *wrappedBody) Read(b []byte) (int, error) {
 	case nil:
 		// nothing to do here but fall through to the return
 	case io.EOF:
-		wb.span.End()
+		wb.span.End() // TODO: why do we need to end the span twice?
```

Comment:
We don't need to write the end-span twice, maybe one is for the end-span and the other is for the start-span? Also, are you sure that it happens twice?
Anyway we try to avoid "TODO" comments, if there's an issue then we can open a ticket about it.

Reply:
we  will solve this in the next commit set